### ISSUE TITLE
[Go] Implement new Concept Exercise: comments

### ISSUE THREAD

**Issue by tehsphinx** [exercism/v3#843]
This issue describes how to implement the `comments` concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the rules about comments in Go.

## Learning objectives

- exported identifiers should have a comment
- Comments should be whole sentences
- Function comments start with the function name (no colon after function name but complete sentence)
- Package comments start with `package XY`
- `golint` to check for missing and wrongly formatted comments

## Out of scope

- gofmt and formatting code

## Concepts

- `comments`

## Prerequisites

None

## Resources to refer to

- https://dave.cheney.net/practical-go/presentations/qcon-china.html#_comments
- https://github.com/golang/go/wiki/CodeReviewComments
- https://blog.golang.org/godoc-documenting-go-code

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by nikimanoledaki** [comment]
Hi, I would love to take this one on! It would be my very first v3 and open-source issue 🎉 It seems to me like a good level to start at.

**Comment by tehsphinx** [comment]
@nikimanoledaki: That'd be awesome!

**Comment by oanaOM** [comment]
Everything seems to be done here.
